app/models/person.rb

```diff
@@ -1,10 +1,24 @@
 class Person < ApplicationRecord
-  belongs_to :user, inverse_of: :person
+  belongs_to :user, optional: true, inverse_of: :person
+  belongs_to :service_area, optional: true
+  belongs_to :location, optional: true
+
+  has_many :asks_as_receiver, class_name: "Listing::Ask", foreign_key: "person_id", inverse_of: :person
+  has_many :offers_as_provider, class_name: "Listing::Offer", foreign_key: "person_id", inverse_of: :person
```

Comment:
Aren't these going to have to filter by `listings.type` somehow?

Also wondering if they can't be named simply `asks` and `offers`?

Also (!), i don't think we're namespacing `Ask` and `Offer` under `Listings::`.

Reply:
Yup, ok. We didn't have validation on `type,` in the model so will add that now.

And, CORRECT. :(